Massacre the build script (#1024)
neow3j/neow3j: RELEASE.md

This file describes the required steps to release neow3j. Before releasing make sure that the external documentation
at [neow3j.io](https://neow3j.io) is up-to-date.

> If this release contains changes for a new Neo release, make sure that the `test-tools` module is equipped 
> with an updated neo-express version. This requires an update of the `neow3j-test-docker` Docker image. 
> See section [Updating neow3j-test-docker](#updating-neow3j-test-docker) for more details. It is possible that 
> at the time of release there is no new `neo-express` version available that supports the new Neo version. 

## Releasing with GitHub Actions

1. Create a branch `release` from `main`.
2. Set the release version number in:
   - build.gradle, in the `version` property
   - Compiler.java, in the `COMPILER_NAME` constant
   - README.md, in the sections that explain how to import neow3j
3. Tag the `main` branch at the **top** commit. Tag format, e.g., `3.14.0`. Note, that the release workflow currently 
   doesn't support releasing anything else than the top commit of a branch.
4. Run the "_Release SDK, devpack, Gradle plugin_" GitHub workflow on the `main` branch. It will publish the packages

>    compiler, contract, core, devpack, devpack-test, test-tools to Maven Central via Sonatype and the Gradle plugin to

Reviewer: ```suggestion
   `compiler`, `contract`, `core`, `devpack`, `devpack-test`, `test-tools` to Maven Central via Sonatype and the `gradle-plugin` to
```

This file describes the required steps to release neow3j. Before releasing make sure that the external documentation
at [neow3j.io](https://neow3j.io) is up-to-date.

> If this release contains changes for a new Neo release, make sure that the `test-tools` module is equipped 
> with an updated neo-express version. This requires an update of the `neow3j-test-docker` Docker image. 
> See section [Updating neow3j-test-docker](#updating-neow3j-test-docker) for more details. It is possible that 
> at the time of release there is no new `neo-express` version available that supports the new Neo version. 

> > For instuctions on how to produce snapshot versions or release locally see DEVLOPMENT.md

Reviewer: ```suggestion
> For instructions on how to produce snapshot versions or release locally see `DEVELOPMENT.md`.
```

The following properties are required to create a release:
- `signingKey`, the PGP key used to sign neow3j artifacts, ASCII armored and encrypted
- `signingPassword`, the password for the PGP key
- `sonatypeUsername`, the username for the Sonatype account, should be a token and not the actual username
- `sonatypePassword`, the password for the Sonatype account, should be a token and not the actual password 
- `gradle.publish.key`, the API key for the Gradle Plugin Repository
- `gradle.publish.secret`, the API secret for the Gradle Plugin Repository

You can set those properties in the `gradle.properties` file in the project root. Or via in the command line with a 
`-P` prefix, e.g., `-PsigningKey=...`.

Putting the armored PGP key in the `gradle.properties` file is a bit ugly but doable. You have to add `\n\` at the 
end of every line - also on empty lines - to make it a valid property value. It will look something like this:
```
signingKey=-----BEGIN PGP PRIVATE KEY BLOCK-----\n\
\n\
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxa0W8BM\n\
...
xxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxxx9e\n\
xxx9E\n\
-----END PGP PRIVATE KEY BLOCK-----

```

> To publish the SDK and devpack artifacts from the currently checked out code base run:

Reviewer: ```suggestion
To publish the SDK and devpack artifacts from the currently checked-out code base, run:
```

```bash
./gradlew --info -x :gradle-plugin:publishToSonatype -x :neofs:publishToSonatype  publishToSonatype closeAndReleaseSonatypeStagingRepository
```
The task `publishToSonatype` compiles, packages, signs and uploads the artifacts to Sonatype into a "Staging
Repository". The task `closeAndReleaseSonatypeStagingRepository` closes and releases that Staging Repository to 
Maven Central. We don't want to release all subprojects to Maven Central. Thus, you can exclude them from the 
command with the `-x` flag[^1].

To publish the Gradle plugin from your currently checked out code base run:
```bash
./gradlew :gradle-plugin:publishPlugin 
```

> ## After Releasing the Artefacts

Reviewer: ```suggestion
## After Releasing the Artifacts
```
